### DISCUSSION TITLE
Get TPS of server

### DISCUSSION THREAD

**Discussion by onLaait** [Minestom/Minestom#1684] [Help] [answered]
Hi, how can i get TPS of server?

**Comment by ZakShearman** [comment, marked as answer]
You can use the ServerTickMonitorEvent to get a TickMonitor and get the time of the last tick.

You can use this over time to calculate the average over different periods. I myself just store the last and display it, but I will average it in the future: https://github.com/towerdefence-cc/minestom-core/blob/main/src/main/java/dev/emortal/minestom/core/module/core/command/PerformanceCommand.java#L58

The TPS in the above code technically isn't actually the TPS as you would need to average over the past second, it's technically just a projection of what the TPS would be if the past 20 ticks were the same as the past 1.

**Comment by dev-hydrogen** [comment]
If youre looking for the target TPS then just use MinecraftServer.TICK_PER_SECOND (TPS is not necessarily always 20 on minestom)
If youre looking for the actual TPS then use ZakShearman's solution
